File: CPP/Car.cpp
/* Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 3 gatau 4 DPBO Composite dalam mata
   kuliah Strukdat untuk keberkahanNya maka saya tidak
   melakukan kecurangan seperti yang telah dispesikasikan.Aamiin*/
#pragma once//kayak requiered once di php
//library
#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

// class untuk mobil subclass dari kendaraan
class Car : public Vehicle
{
private://property
    int jumlahKursi;
    int jumlahPintu;

public:
    //constructor default untuk mobil dibawahnya constructor overload karena tidak memungkinkan untuk null pada data jadi di defaultkan dan ditambah parameter dari superclass
    Car()
    {
        this->jumlahKursi = 0;
        this->jumlahPintu = 0;
    }

    Car(int jumlahKursi, string platNomor, string merk, int tahunProduksi, string warna) : Vehicle(platNomor, merk, tahunProduksi, warna){
        this->jumlahKursi = jumlahKursi;
        this->jumlahPintu = 0;
    }

    Car(int jumlahKursi, int jumlahPintu, string platNomor, string merk, int tahunProduksi, string warna) : Vehicle(platNomor, merk, tahunProduksi, warna){
        this->jumlahKursi = jumlahKursi;
        this->jumlahPintu = jumlahPintu;
    }
    //getetr dan setter
    void setjumlahKursi(int jumlahKursi){
        this->jumlahKursi = jumlahKursi;
    }

    int getjumlahKursi(){
        return this->jumlahKursi;
    }

    void setJumlahPintu(int jumlahPintu){
        this->jumlahPintu = jumlahPintu;
    }

    int getJumlahPintu(){
        return this->jumlahPintu;
    }
    //destructor
    ~Car(){

    }
};

File: PYTHON/Car.py
# /* Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 3 gatau 4 DPBO Composite dalam mata
#    kuliah OOP untuk keberkahanNya maka saya tidak
#    melakukan kecurangan seperti yang telah dispesikasikan.Aamiin*/

from Vehicle import Vehicle

# class untuk mobil subclass dari kendaraan/Vehicle
class Car(Vehicle):
    # constructor
    def __init__(self, jumlahKursi=0, jumlahPintu=0, platNomor="", merk="", tahunProduksi=0, warna=""):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        # property mobil ada jumlah kursi dan jumlah pintu
        self.__jumlahKursi = jumlahKursi
        self.__jumlahPintu = jumlahPintu

    # Getter dan setter method untuk jumlahKursi
    def setJumlahKursi(self, jumlahKursi):
        self.__jumlahKursi = jumlahKursi

    def getJumlahKursi(self):
        return self.__jumlahKursi

    # Getter dan setter method untuk jumlahPintu
    def setJumlahPintu(self, jumlahPintu):
        self.__jumlahPintu = jumlahPintu

    def getJumlahPintu(self):
        return self.__jumlahPintu
